### PR TITLE
fix: support specifying release per repo for download

### LINKED ISSUE #2490: Issues with multi model downloads
**Describe the bug**
Currently `ilab model download` supports specifying multiple repositories and filenames to be downloaded via `--repository` and `--filenames` 
however, 
- `--release` still only accepts a single value, forcing users to download the same version for all specified models whether it exists or not
- there is an assumption that users are either specifying all safetensor repos or all GGUF repos. If users were to try and specify a mix of repositories and also specify a list of filenames that don't match one-to-one with the repositories, the workflow breaks and models are not downloaded properly. Users also don't have a way to express which filenames should be mapped to which specified repository 

**To Reproduce**
1. run `ilab model download -repository instructlab/granite-7b-lab -repository instructlab/granite-7b-lab-GGUF -filename granite-7b-lab-Q4_K_M.gguf`
2. See download not proceeding as expected 

**Expected behavior**
All specified models are downloaded as expected 

**Screenshots**

**Device Info (please complete the following information):**
```
╰─ ilab system info                                                                                                                                                                        ─╯
Platform:
  sys.version: 3.11.9 (v3.11.9:de54cf5be3, Apr  2 2024, 07:12:50) [Clang 13.0.0 (clang-1300.0.29.30)]
  sys.platform: darwin
  os.name: posix
  platform.release: 24.0.0
  platform.machine: arm64
  platform.node: jrao-mac
  platform.python_version: 3.11.9
  platform.cpu_brand: Apple M3 Pro
  memory.total: 36.00 GB
  memory.available: 14.38 GB
  memory.used: 15.94 GB

InstructLab:
  instructlab.version: 0.19.0rc2.dev154
  instructlab-dolomite.version: 0.1.1
  instructlab-eval.version: 0.3.1
  instructlab-quantize.version: 0.1.0
  instructlab-schema.version: 0.4.1
  instructlab-sdg.version: 0.3.0
  instructlab-training.version: 0.5.2

Torch:
  torch.version: 2.3.1
  torch.backends.cpu.capability: NO AVX
  torch.version.cuda: None
  torch.version.hip: None
  torch.cuda.available: False
  torch.backends.cuda.is_built: False
  torch.backends.mps.is_built: True
  torch.backends.mps.is_available: True

llama_cpp_python:
  llama_cpp_python.version: 0.2.79
  llama_cpp_python.supports_gpu_offload: True
```

**Additional context**

### PR DESCRIPTION
**Pull request by jaideepr97**
allows specifying release to be downloaded per repo

addresses #2490 partially

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the
  [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary).
- [ ] [Changelog](https://github.com/instructlab/instructlab/blob/main/CHANGELOG.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Functional tests have been added, if necessary.
- [ ] E2E Workflow tests have been added, if necessary.

### CODE REVIEW

**Comment by nathan-weinberg** [comment]
Since we're mixing and matching now should we add coverage to `test_download`?

**Comment by jaideepr97** [comment]
> Member

this doesn't allow mixing and matching
it only fixes the `--release` bug

IMO we need a better design to allow mixing and matching properly

File: src/instructlab/model/download.py [hunk 1]
```diff
@@ -264,7 +265,13 @@ def download(self):
 )
 @click.option(
     "--release",
-    default="main",  # TODO: add to config.yaml
+    "-rl",
```
Comment (cdoern) [inline src/instructlab/model/download.py:268]: shorthands are nice, thanks

**Review by cdoern — APPROVED** [review]
thanks for the quick fix, but yes we should scope adding testing for all of this

**Review by nathan-weinberg — APPROVED** [review]

**Comment by nathan-weinberg** [comment]
@Mergifyio backport release-v0.19

**Comment by cdoern** [comment]
note, this backport needs to merge AFTER the one for #2487

File: src/instructlab/model/download.py [hunk 2]
```diff
@@ -264,7 +265,13 @@ def download(self):
...
+    "releases",
+    multiple=True,
+    default=[
+        "main",
+        "main",
```
Comment (alinaryan) [inline src/instructlab/model/download.py:273]: do we need 'main' twice here?

Comment (jaideepr97) [reply src/instructlab/model/download.py:273]: we need 2 release entries for the 2 default models we are downloading by default now
there are better ways to do this but this seemed like the most straightforward for what is being pursued, anything more felt like going out of the way to make it more complicated

**Review by alinaryan — APPROVED** [review]

**Pull request merged**